presto-common/src/main/java/com/facebook/presto/common/block/ArrayBlockEncoding.java

```diff
@@ -62,7 +62,7 @@ public Block readBlock(BlockEncodingSerde blockEncodingSerde, SliceInput sliceIn
         int positionCount = sliceInput.readInt();
         int[] offsets = new int[positionCount + 1];
         sliceInput.readBytes(Slices.wrappedIntArray(offsets));
-        boolean[] valueIsNull = decodeNullBits(sliceInput, positionCount).orElseGet(() -> new boolean[positionCount]);
+        boolean[] valueIsNull = decodeNullBits(sliceInput, positionCount).orElse(null);
```

Comment:
lol does `null` valueIsNull means all-false? cc: @yingsu00. If it is the case, can enrich the javadoc for `createArrayBlockInternal`? Same for `createRowBlockInternal`.

Reply:
Certainly the methods seem to handle it that way (although some methods like `AbstractRowBlock#copyPositions` will unconditionally create a `boolean[] newValueIsNull` unnecessarily), and the method and fields are all marked `@Nullable`